The one thing we changed between 0.9.12 and 0.9.13.

diff --git a/src/hmac.c b/src/hmac.c
index 839f4b1..c8ef66e 100644
--- a/src/hmac.c
+++ b/src/hmac.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2008,2009 Red Hat, Inc.
+ * Copyright 2008,2009,2013 Red Hat, Inc.
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions
@@ -528,7 +528,7 @@ main(int argc, char **argv)
 {
 	SECStatus status;
 	FILE *fp;
-	int i, c, ret, Pflag, qflag, sflag, bflag, truncate;
+	int i, c, ret, Pflag, qflag, sflag, bflag, truncate, checked;
 	size_t n;
 	unsigned int u;
 	struct hashinfo *algorithm, *default_algorithm;
@@ -727,7 +727,8 @@ main(int argc, char **argv)
 			exit(-1);
 		}
 	} else {
-		if ((copt == NULL) && (optind == argc)) {
+		if (((copt == NULL) && (optind == argc)) ||
+		    ((copt != NULL) && (optind != argc))) {
 			fprintf(stderr, "%s", usage);
 			exit(-1);
 		}
@@ -977,6 +978,7 @@ main(int argc, char **argv)
 		} else {
 			/* Iterate through the files named by the input file. */
 			ret = 0;
+			checked = 0;
 			while (fgets(cbuf, sizeof(cbuf), fp) != NULL) {
 				/* Skip up to the first non-whitespace. */
 				expected = cbuf;
@@ -1033,6 +1035,7 @@ main(int argc, char **argv)
 				    (truncate < (algorithm->digestsize * 8))) {
 					truncate_hex(hex_result, truncate);
 				}
+				checked++;
 				if (strcasecmp(expected, hex_result) != 0) {
 					if (!qflag) {
 						fprintf(stderr, "%s: FAILED\n"
@@ -1053,6 +1056,12 @@ main(int argc, char **argv)
 			if (fp != stdin) {
 				fclose(fp);
 			}
+			if (checked == 0) {
+				fprintf(stderr, "%s: no properly formatted %s "
+					"checksum lines found\n", copt,
+					default_algorithm->hash);
+				ret = -1;
+			}
 		}
 	}
 
